### PR TITLE
surge metrics integration with influx and micrometer

### PR DESCRIPTION
**Pull request by satyender071**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -80,7 +80,13 @@ lazy val `surge-engine-command-javadsl` =
 lazy val `surge-engine-multilanguage` =
   (project in file("modules/multilanguage"))
     .dependsOn(`surge-engine-command-scaladsl`)
-    .settings(libraryDependencies ++= Seq(Akka.slf4j, logback, slf4jApi))
+    .settings(libraryDependencies ++= Seq(Akka.slf4j, logback, slf4jApi,
+      "com.typesafe.akka" %% "akka-protobuf-v3" % "2.6.16",
+      "com.typesafe.akka" %% "akka-pki" % "2.6.16",
+      "com.typesafe.akka" %% "akka-discovery" % "2.6.16",
```
Comment (jeffboutotte) [inline build.sbt:86]: It doesn't look like these are actually used, but if so can you pull them into `project/Dependencies.scala` and version them with that same `Akka.version` variable to keep all of those consistent more easily?

File: build.sbt [hunk 2]
```diff
@@ -80,7 +80,13 @@ lazy val `surge-engine-command-javadsl` =
...
+      "io.micrometer" % "micrometer-registry-influx" % "1.7.3",
+      "com.ukg" %% "surge-micrometer-binder" % "0.5.19-SNAPSHOT"
```
[the remark this answers is not in the document]
Comment (sebastianharko) [reply build.sbt:88]: @jeffboutotte Yes you are correct, we are just trying to see if it works, we did a publishLocal.... So this PR is probably not going to get merged as is , it's more like a POC

File: modules/multilanguage/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+1 - Provide the influxdb config 
+
+2 - create InfluxMeterRegistry
+
+3 - Bind your meter registry to your metric binder implementation
+
+below is an example for reference : - 
+
+    val config: InfluxConfig = new InfluxConfig() {
```
Comment (sebastianharko) [inline modules/multilanguage/README.md:13]: @satyender071 use markdown syntax highlighting ;-) (https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks)

Comment (satyender071) [reply modules/multilanguage/README.md]: Added.

File: modules/multilanguage/src/main/scala/com/ukg/surge/multilanguage/GenericAsyncAggregateCommandModel.scala [hunk 4]
```diff
@@ -42,53 +43,60 @@ class GenericAsyncAggregateCommandModel(bridgeToBusinessApp: BusinessLogicServic
     }
   }
 
-  override def processCommand(aggregate: Option[SurgeState], surgeCommand: SurgeCmd): Future[Seq[SurgeEvent]] = {
-    if (!validIds(aggregate, surgeCommand)) {
-      Future.failed(new Exception("Wrong aggregate ids!"))
-    } else {
-      logger.info(
-        s"Calling command handler of business app via gRPC. Aggregate id: ${surgeCommand.aggregateId}). State defined: ${aggregate.isDefined}. Command payload size: ${surgeCommand.payload.length} (bytes).")
+  val metric: Metrics = Metrics.globalMetricRegistry
```
Comment (sebastianharko) [inline modules/multilanguage/src/main/scala/com/ukg/surge/multilanguage/GenericAsyncAggregateCommandModel.scala:46]: can be made private

File: modules/multilanguage/src/main/scala/com/ukg/surge/multilanguage/GenericAsyncAggregateCommandModel.scala [hunk 5]
```diff
@@ -42,53 +43,60 @@ class GenericAsyncAggregateCommandModel(bridgeToBusinessApp: BusinessLogicServic
...
+  val processCommandMetric: Timer = metric.timer(MetricInfo("ProcessCommand", "Process Command Metric"))
```
Comment (sebastianharko) [inline modules/multilanguage/src/main/scala/com/ukg/surge/multilanguage/GenericAsyncAggregateCommandModel.scala:47]: can be made private as well

File: modules/multilanguage/src/main/scala/com/ukg/surge/multilanguage/GenericAsyncAggregateCommandModel.scala [hunk 6]
```diff
@@ -42,53 +43,60 @@ class GenericAsyncAggregateCommandModel(bridgeToBusinessApp: BusinessLogicServic
...
   }
 
-  override def processCommand(aggregate: Option[SurgeState], surgeCommand: SurgeCmd): Future[Seq[SurgeEvent]] = {
-    if (!validIds(aggregate, surgeCommand)) {
-      Future.failed(new Exception("Wrong aggregate ids!"))
-    } else {
-      logger.info(
-        s"Calling command handler of business app via gRPC. Aggregate id: ${surgeCommand.aggregateId}). State defined: ${aggregate.isDefined}. Command payload size: ${surgeCommand.payload.length} (bytes).")
+  private val metric: Metrics = Metrics.globalMetricRegistry
+  private val processCommandMetric: Timer = metric.timer(MetricInfo("ProcessCommand", "Process Command Metric"))
```
Comment (jeffboutotte) [inline modules/multilanguage/src/main/scala/com/ukg/surge/multilanguage/GenericAsyncAggregateCommandModel.scala:47]: Can you make the description a bit more descriptive here and scope the metric name a bit as well? Maybe something like `surge.grpc.process-command-timer` for the name.

File: modules/multilanguage/src/main/scala/com/ukg/surge/multilanguage/MultilanguageGatewayServiceImpl.scala [hunk 7]
```diff
@@ -42,7 +42,10 @@ class MultilanguageGatewayServiceImpl(aggregateName: String, eventsTopicName: St
     engine
   }
 
-  override def forwardCommand(in: ForwardCommandRequest): Future[ForwardCommandReply] = {
+  private val metric: Metrics = Metrics.globalMetricRegistry
+  private val forwardCommandMetric: Timer = metric.timer(MetricInfo("ForwardCommand", "Forward Command Metric"))
```
Comment (jeffboutotte) [inline modules/multilanguage/src/main/scala/com/ukg/surge/multilanguage/MultilanguageGatewayServiceImpl.scala:46]: Same comment here and for the getStateMetric below for the metric name/description

File: modules/multilanguage/src/main/scala/com/ukg/surge/multilanguage/GenericAsyncAggregateCommandModel.scala [hunk 8]
```diff
@@ -42,53 +43,60 @@ class GenericAsyncAggregateCommandModel(bridgeToBusinessApp: BusinessLogicServic
...
-  }
 
-  override def handleEvents(aggregate: Option[SurgeState], surgeEvents: Seq[SurgeEvent]): Future[Option[SurgeState]] = {
-    if (surgeEvents.isEmpty) {
-      logger.warning("handleEvents called but no events provided!")
-      Future.successful(aggregate)
-    } else {
-      if (!validIds(aggregate, surgeEvents)) {
-        Future.failed(new Exception("handleEvents called but wrong aggregate ids!"))
+  private val handleEventsMetric: Timer = metric.timer(MetricInfo("HandleEvents", "Handle Events Metric"))
```
Comment (jeffboutotte) [inline modules/multilanguage/src/main/scala/com/ukg/surge/multilanguage/GenericAsyncAggregateCommandModel.scala:79]: This name/description for the metric might want to be a bit more descriptive as well.

**Review by jeffboutotte — COMMENTED** [review]
Few minor things, but looks good overall!

File: modules/multilanguage/src/main/scala/com/ukg/surge/multilanguage/GenericAsyncAggregateCommandModel.scala (hunk 6 above)
Comment (satyender071) [reply modules/multilanguage/src/main/scala/com/ukg/surge/multilanguage/GenericAsyncAggregateCommandModel.scala]: done.

File: modules/multilanguage/src/main/scala/com/ukg/surge/multilanguage/MultilanguageGatewayServiceImpl.scala (hunk 7 above)
Comment (satyender071) [reply modules/multilanguage/src/main/scala/com/ukg/surge/multilanguage/MultilanguageGatewayServiceImpl.scala]: Changed

File: modules/multilanguage/src/main/scala/com/ukg/surge/multilanguage/GenericAsyncAggregateCommandModel.scala (hunk 8 above)
Comment (satyender071) [reply modules/multilanguage/src/main/scala/com/ukg/surge/multilanguage/GenericAsyncAggregateCommandModel.scala]: changed.

**Review by jeffboutotte — APPROVED** [review]

**Pull request merged**